rotkehlchen/chain/evm/decoding/cctp/decoder.py

```diff
@@ -0,0 +1,106 @@
+
+    def _maybe_enrich_cctp_transfers(self, context: EnricherContext) -> TransferEnrichmentOutput:
+        """Enriches CCTP transfers and withdrawals."""
+        if (
+            context.event.event_type == HistoryEventType.SPEND and
+            context.event.event_subtype == HistoryEventSubType.NONE and
+            context.event.asset.identifier == self.asset_identifier and
+            context.event.address == self.token_minter
+        ):
+            chain_info = ''
+            for txn in context.all_logs:
+                if txn.topics[0] == DEPOSIT_FOR_BURN:
+                    to_chain = hex_or_bytes_to_int(txn.data[64:96])
+                    try:
+                        chain_info = f' from {self.evm_inquirer.chain_id.label()} to {CCTP_DOMAIN_MAPPING[to_chain].label()}'  # noqa: E501
+                    except KeyError:
+                        log.error(f'Could not find chain ID {to_chain} for CCTP transfer from {self.evm_inquirer.chain_name}')  # noqa: E501
+                    break
+
+            context.event.event_type = HistoryEventType.DEPOSIT
+            context.event.event_subtype = HistoryEventSubType.BRIDGE
+            context.event.notes = f'Bridge {context.event.balance.amount} USDC{chain_info} via CCTP'  # noqa: E501
+            context.event.counterparty = CPT_CCTP
+            return TransferEnrichmentOutput(matched_counterparty=CPT_CCTP)
+
+        elif (
+            context.event.event_type == HistoryEventType.RECEIVE and
+            context.event.event_subtype == HistoryEventSubType.NONE and
+            context.event.asset.identifier == self.asset_identifier and
+            context.event.address == ZERO_ADDRESS
+        ):
+            chain_info = ''
+            for txn in context.all_logs:
+                if txn.topics[0] == MESSAGE_RECEIVED:
+                    from_chain = hex_or_bytes_to_int(txn.data[:32])
+                    try:
+                        chain_info = f' from {CCTP_DOMAIN_MAPPING[from_chain].label()} to {self.evm_inquirer.chain_id.label()}'  # noqa: E501
+                    except KeyError:
+                        log.error(f'Could not find chain ID {from_chain} for CCTP transfer to {self.evm_inquirer.chain_name}')  # noqa: E501
+                    break
+
+            context.event.event_type = HistoryEventType.WITHDRAWAL
+            context.event.event_subtype = HistoryEventSubType.BRIDGE
+            context.event.notes = f'Bridge {context.event.balance.amount} USDC{chain_info} via CCTP'  # noqa: E501
+            context.event.counterparty = CPT_CCTP
+            return TransferEnrichmentOutput(matched_counterparty=CPT_CCTP)
+
+        return FAILED_ENRICHMENT_OUTPUT
+
+    # -- DecoderInterface methods
+
+    @ staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (CCTP_CPT_DETAILS,)
+
+    def enricher_rules(self) -> list[Callable]:
+        return [self._maybe_enrich_cctp_transfers]
```

Comment:
why is this a transfer enricher? Don't all the transactions emit a `DEPOSIT_FOR_BURN` from certaina address?

Reply:
Same for the receive. There is a `MintAndWithdraw` that you can match. Can't you?

Reply:
Yes, moved the decoders to `addresses_to_decoders`